p2p/rlpx.go

```diff
@@ -242,106 +266,77 @@ func (h *encHandshake) ecdhShared(prv *ecdsa.PrivateKey) ([]byte, error) {
 // it should be called on the dialing side of the connection.
 //
 // prv is the local client's private key.
-// token is the token from a previous session with this node.
 func initiatorEncHandshake(conn io.ReadWriter, prv *ecdsa.PrivateKey, remoteID discover.NodeID, token []byte) (s secrets, err error) {
-	h, err := newInitiatorHandshake(remoteID)
+	h := &encHandshake{initiator: true, remoteID: remoteID}
+	authMsg, err := h.makeAuthMsg(prv, token)
 	if err != nil {
 		return s, err
 	}
-	auth, err := h.authMsg(prv, token)
+	var authPacket []byte
+	if os.Getenv("RLPX_EIP8") != "" {
```

Comment:
Yes, it's more like the latter comment. I used the env variable for testing with other clients.
Would be nice to ensure that the majority of the network will send EIP-8 soon-ish, maybe through
something like 

```go
if time.Now().After(switchoverDate) {
    // send EIP-8
}
```